docs/guide/writing-migrations.md

```diff
@@ -0,0 +1,302 @@
+# Writing Migrations
+
+Migrations in ApostropheCMS allow you to make targeted changes to your database, ensuring that your data stays in sync with the evolving structure of your code. They are particularly useful when new code deployments introduce changes to the underlying data model or schema, such as adding new fields, removing deprecated properties, or correcting inconsistencies in existing content. In this guide, we will walk through how to write migrations to add or remove properties from existing pieces and widgets.
+
+## Adding migrations
+
+In ApostropheCMS, migrations are added using the [`add(name, fn)` method](/reference/modules/migration.md#add-name-fn) of the `@apostrophecms/migration` module. One common place to add these are within the `init(self)` initialization function of your module. Each migration requires a unique name and is only run **once**. ApostropheCMS tracks which migrations have already been executed, ensuring they won’t run again across restarts or deployments.
+
+While the migration function can be added as an anonymous function as the second argument to the `add()`method, they can also be defined in the `methods(self)` customization function of the module. This can provide for a cleaner `init(self)` function, but is a matter of preference.
+
+Example adding the migration to `init(self)`:
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  async init(self) {
+    // Adding a migration named 'add-featured-to-articles'
+    self.migration.add('add-featured-to-articles', async () => {
+      return self.apos.migration.eachDoc({
+        type: 'article'
+      }, async (doc) => {
+        if (doc.featured === undefined) {
```

Comment:
I think a simple example like this is still good, but I would document it as a way to clean up if a string field was first added with no `def` and then you decide, as a one-time thing, to change the empty string to a more useful default later. Apostrophe can't do that automatically because it can't divine your intent and the empty string can be a legitimate value.

Reply:
Changed.